src/models/connectionStore.ts

```diff
@@ -459,6 +457,14 @@ export class ConnectionStore {
 		// Connections defined in workspace scope are unioned with the Connections defined in user scope
 		let profilesInConfiguration = this._connectionConfig.getConnections(true);
 
+		// Update encrypt property value for each profile
+		profilesInConfiguration.forEach(async profile => {
```

Comment:
FYI - this currently is going to continue on without awaiting for all the profiles to be updated, which probably isn't what you want. You should be doing `await Promise.all(profilesInConfiguration.map(async profile => ...` to do that.

Reply:
I also have a limitation that 'loadAllConnections' is a non-async function.
Should I do `Promise.all(...).then(() => { });` in that case? Any other way of calling async function synchronously?

Reply:
No, unfortunately there's no way to do that in TS (wish there was a .Wait() like C#...)

So you'd have to make all calling functions async to fully fix it.

Reply:
I also noticed profilesConfiguration.map would proceed in parallel and causes inconsistent behavior.. I did make the method async and also used for loop to update 1 profile at a time.